contracts/farms/common/RevoFarmBot.sol

```diff
@@ -3,25 +3,38 @@ pragma solidity 0.8.4;
 
 import "hardhat/console.sol";
 
-import "./IMoolaStakingRewards.sol";
-import "./ubeswap/contracts/uniswapv2/interfaces/IUniswapV2Router02.sol";
-import "./ubeswap/contracts/uniswapv2/interfaces/IUbeswapRouter.sol";
-import "./ubeswap/contracts/uniswapv2/interfaces/IUniswapV2Pair.sol";
-import "./IRevoFees.sol";
-import "./openzeppelin-solidity/contracts/ERC20.sol";
-import "./openzeppelin-solidity/contracts/AccessControl.sol";
-import "./openzeppelin-solidity/contracts/SafeERC20.sol";
-import "./openzeppelin-solidity/contracts/Pausable.sol";
-
-contract UbeswapFarmBot is ERC20, AccessControl, Pausable {
+import "../../openzeppelin-solidity/contracts/ERC20.sol";
+import "../../openzeppelin-solidity/contracts/AccessControl.sol";
+import "../../openzeppelin-solidity/contracts/SafeERC20.sol";
+import "../../openzeppelin-solidity/contracts/Pausable.sol";
+import "../../fees/interfaces/IRevoFees.sol";
+
+/**
+ * RevoFarmBot is an abstract base class that can be used to implement Revo farm bots on top
+ * of arbitrary sources of yield. These sources of yield come from staking some ERC20 token
+ * (the "staking token") into a contract.
+ *
+ * Minimally, subclasses must implement the _deposit, _withdraw, and _claimRewards methods that
+ * deposit/withdraw an amount of the staked token into the "farm", and claim rewards. This will
+ * result in a RevoFarmBot that does not auto-compound yield; it acts only as "storage" for a
+ * user's staking token.
+ *
+ * Notably, this class does not contain a virtual "compound" function; this is because depending
+ * on the implementation, a function repsonsible for compounding the farm bot's staked position may
+ * require different arguments from the caller.
```

Comment:
Sure. Single staking farms are of course an example (in particular the UBE single staking on Ubeswap), since we'll either need to do only a single swap, or none at all. 

I have not dug extremely deep into it, but if you want a more substantive example, the function signature Symmetric uses for swaps is considerably different than the one used in the Uniswap protocol. See `multihopBatchSwapExactIn` [here](https://explorer.celo.org/address/0x7EbE42906a5F1DD96E47E7152102A6aA0A0d2459/write-contract). This is the method that gets called when using the Symmetric's dApp to swap. `swapSequences` is actually a list of lists of structs, each of which contains six separate fields (note that this is a list of lists of structs in order to do a _single_ swap, where we currently need just a list of addresses for a single swap).

Mobius's interface for swapping is different too. See `swap` [here](https://explorer.celo.org/address/0x0986B42F5f9C42FeEef66fC23eba9ea1164C916D/write-contract). It doesn't require a  path, or even start and end token addresses, but a `uint8` index into what I assume is some sort of list of registered tokens. Again, haven't looked too deep into it, but my general impression is that it is somewhat of a coincidence that both Sushi and Ubeswap share the same underlying Uniswap routing interface.